xilinx/common/tcl/prologue.tcl

```diff
@@ -94,6 +94,10 @@ proc load_vsrc_manifest {obj vsrc_manifest} {
       lappend relative_files [file join [file dirname $vsrc_manifest] $path]
     }
   }
+  if {[info exists ::env(WAKE_IP_RESOURCE_FILES)]} {
```

Comment:
This feels a little bit hacky to me because we're forcing this TCL script to understand something about an environment variable that Wake is setting. In terms of separation of concerns, this feels like it's mixing concerns that this script should not have to be aware of. Is there somewhere to add a layer of indirection such that something _else_ is injecting the extra files?

One idea: Could the Wake build rule for Vivado builds serialize the list of files into a `.f` (or `.F`) file that refers to the Verilog files? Just note that Vivado doesn't natively support this `.F` file; it was myself that implemented the function that parses the `.F` file passed to this script, so it doesn't support the full set of semantics of `.F` files.

Reply:
it would have to be done in a wrapper script that also invokes vivado. like we do with verilator. since the environment variable value isn't visible to wake

Reply:
The exists will return false if it doesnt exist, so it wont be an issue, but I see your concern. 

This actually is essentially sidestepping the .F file. If we want to add this to the .F file, then wake needs to know about it, which means having wake call a shell script to read the environment variable, and append that to the generated .F. Albert seemed to think this was a better alternative

Reply:
Yes, I agree that it would have to be done in the wrapper script, and I agree that this script has the correct behavior in the non-Wake cases where this environment variable wouldn't be set. However, I still stand by my comment that by having this script read directly from this environment variable, we're mixing up different layers of concerns of the program. There really should only be one way of passing Verilog files into this script, and if we need a way of wrapping file paths that come dynamically from the Wake environment, then I think that wrapper script is necessary, at least when invoked from Wake.

Reply:
Yeah i don't disagree with you there. I will look into doing it the other way